src/hash_table.py

```diff
@@ -0,0 +1,31 @@
+# _*_ encoding: utf-8 _*_
+"""Simple implementation of a hash table."""
+
+
+class HashTable(object):
+    """Sample Python Hash Table."""
+
+    def __init__(self, slots):
+        """Init method for hash table."""
+        self.slots = slots
+        self._table = [[] for num in range(slots)]
+
+    def _hash(self, key):
+        """Render key string as integer with additive algorithm."""
+        if not isinstance(key, str):
+            raise TypeError('please give us a string')
+        hashnum = sum(ord(char) for char in key)
+        return hashnum % self.slots
+
+    def set(self, key, value):
+        """Store given value in table using given key."""
+        hashed_key = self._hash(key)
+        self._table[hashed_key].append((key, value))
```

Comment:
What do you do in the event that you try to re-set a key? All you'd do is add a new tuple to the same bucket. Your newly-set value would never be caught by `get` though, as it'd first find the initially-set value before it found the new value.